pkg/platform/platform.go

```diff
@@ -0,0 +1,24 @@
+package platform
+
+import (
+	"runtime"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/docker/docker/pkg/parsers/architecture"
+)
+
+var (
+	// Architecture holds the runtime architecture of the process.
+	Architecture string
+	// OSType holds the runtime operating system type (Linux, …) of the process.
+	OSType string
+)
+
+func init() {
+	var err error
+	Architecture, err = architecture.GetRuntimeArchitecture()
```

Comment:
@vdemeester sorry i was thinking we remove the `pkg/parsers/architecture` package and just have all that logic in `pkg/platform` here. WDYT?